src/components/errorboundary/errorboundary.jsx

```diff
@@ -9,39 +8,47 @@ class ErrorBoundary extends React.Component {
     constructor (props) {
         super(props);
         this.state = {
-            hasError: false,
-            errorId: null
+            error: null,
+            errorInfo: null
         };
     }
 
+    /**
+     * Handle an error caught by this ErrorBoundary component.
+     * @param {Error} error - the error that was caught.
+     * @param {React.ErrorInfo} errorInfo - the React error info associated with the error.
+     */
     componentDidCatch (error, errorInfo) {
-        // Display fallback UI
-        Sentry.withScope(scope => {
-            scope.setTag('project', 'scratch-www');
-            if (this.props.componentName) {
-                scope.setTag('component', this.props.componentName);
-            }
-            Object.keys(errorInfo).forEach(key => {
-                scope.setExtra(key, errorInfo[key]);
+        error = error || {
+            stack: 'Unknown stack',
+            message: 'Unknown error'
+        };
+        errorInfo = errorInfo || {
+            componentStack: 'Unknown component stack'
+        };
+
+        // only remember the first error: later errors might just be side effects of that first one
+        if (!this.state.error) {
```

Comment:
Is there ever a case where there would be multiple distinct errors (and it would matter)?

Reply:
I think it's most likely that we'd have just one error: after the first error, the `ErrorBoundary` component will stop rendering its normal children and swap in the `CrashMessageComponent` instead. After that, it seems likely that further errors would either be a problem in `CrashMessageComponent` (which would be REALLY unfortunate, but seems unlikely) or a problem in an async callback or timer or something like that, which wouldn't be caught by `ErrorBoundary` anyway. I'm not 100% confident that those are the only options, which is why the `log.error` reports every error 😅